wasmtest.sh

```diff
@@ -0,0 +1,256 @@
+    fi
+
+    echo
+}
+
+#function to compile all the .c test files in the unit-tests folder
+compile_all_tests() {
+    echo -e "${GREEN}Compiling all test cases in $test_file_base...${RESET}"
+
+    find "$test_file_base" -type f -name "*.c" | while read -r test_file; do
+        test_name=$(basename "$test_file" .c)
+        test_dir=$(dirname "$test_file")
+
+        echo -e "\n${GREEN}Compiling test: $test_name in $test_dir${RESET}"
+
+        compile_single_test "$test_name.c"
+    done
+
+    echo -e "${GREEN}All tests compiled.${RESET}"
+}
+
+#function to run all the .c test files in the unit-tests folder
+run_all_tests() {
+    echo -e "${GREEN}Running all test cases in $test_file_base...${RESET}"
+
+    find "$test_file_base" -type f -name "*.c" | while read -r test_file; do
+        test_name=$(basename "$test_file" .c)
+        test_dir=$(dirname "$test_file")
+
+        echo -e "\n${GREEN}Running test: $test_name in $test_dir${RESET}"
+
+        run_single_test "$test_dir/$test_name"
+    done
+
+    echo -e "${GREEN}All tests completed.${RESET}"
+}
+
+#function to compile all the files from the file user give
+compile_from_files() {
+    local file_list="$1"
+
+    if [ ! -f "$file_list" ]; then
+        echo -e "${RED}Error: File list $file_list not found.${RESET}"
+        exit 1
+    fi
+
+    echo -e "${GREEN}Compiling test cases from file: $file_list...${RESET}"
+
+    while IFS= read -r test_file; do
+        if [ -n "$test_file" ]; then
+            find "$test_file_base" -type f -name "$test_file" 2>/dev/null | while read -r full_path; do
+                test_name=$(basename "$full_path" .c)
+                test_dir=$(dirname "$full_path")
+                compile_single_test "$test_file"
+            done
+        fi
+    done < "$file_list"
+
+    echo -e "${GREEN}All specified tests compiled.${RESET}"
+}
+
+#function to run all the files from the file user give
+run_from_files() {
+    local file_list="$1"
+
+    if [ ! -f "$file_list" ]; then
+        echo -e "${RED}Error: File list $file_list not found.${RESET}"
+        exit 1
+    fi
+
+    echo -e "${GREEN}Running tests from file list: $file_list...${RESET}"
+
+    while IFS= read -r test_file; do
+        if [ -n "$test_file" ]; then
+            find "$test_file_base" -type f -name "$test_file" | while read -r full_path; do
+                test_name=$(basename "$full_path" .c)
+                test_dir=$(dirname "$full_path")
+
+                echo -e "\n${GREEN}Running test: $test_name in $test_dir${RESET}"
+
+                run_single_test "$test_dir/$test_name"
+            done
+        fi
+    done < "$file_list"
+
+    echo -e "${GREEN}All tests from file list completed.${RESET}"
+}
+
+pmode=0
+if [ "${!#}" = "-p" ]; then
+    pmode=1
+
+    # Remove the last argument
+    new_args=("${@:1:$#-1}")
+    # Overwrite the positional parameters with the new arguments
+    set -- "${new_args[@]}"
+fi
+
+case "$1" in
+    filescompile)
```

Comment:
I think it would be a good practice to avoid confusion between singular and plural forms in command-line argument. I believe if you use `filescompile` here, people will sometimes mistakenly type `filecompile` instead and get confused. I believe have both singular and plural form of the command supported is the most user-friendly approach. For example, change this to `filescompile|filecompile`. This applies all other commands below. Besides, have an abbreviated form of the command allowed is also something I prefer.